provision.sh

```diff
@@ -14,31 +14,47 @@ echo "---------------------------------------------"
 echo "Running vagrant provisioning"
 echo "---------------------------------------------"
 
+# install heroku toolbelt
+echo "-------------- Installing heroku toolbelt -------------------------"
+#wget -O- https://toolbelt.heroku.com/install-ubuntu.sh | sh
+# These shell script snippets are directly taken from heroku installation script
+# We want to avoid the apt-get update
+# add heroku repository to apt
+echo "deb http://toolbelt.heroku.com/ubuntu ./" > /etc/apt/sources.list.d/heroku.list
```

Comment:
`apt-get update` does not install anything. That's `apt-get upgrade`.

In general terms, `apt-get update` updates the list of installed packages and dependencies. It hits a lot of package servers and finds out latest information about what new package versions have been released for all installed software.

Then, it goes on and updates the list of dependencies properly on your machine.

Here, `apt-get update` is needed, to update its internal lists and storages to inform of heroku-toolbelt. But we run it once.

Because it's a time consuming operation. In its first run itself, it would pick up things necessary for next installations.